examples/gaussian_process/new/plot_gpc_xor.py

```diff
@@ -0,0 +1,41 @@
+"""Gaussian process classification (GPC)
+
+This example illustrates GPC on XOR data.
+"""
+print __doc__
+
+# Authors: Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
+#
+# License: BSD 3 clause
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn.gaussian_process import GaussianProcessClassification
+from sklearn.gaussian_process.kernels import RBF
+
+xx, yy = np.meshgrid(np.linspace(-3, 3, 50),
+                     np.linspace(-3, 3, 50))
+np.random.seed(0)
+X = np.random.randn(200, 2)
+Y = np.logical_xor(X[:, 0] > 0, X[:, 1] > 0)
+
+# fit the model
+kernel = [1e-10, 1.0, 100] * RBF(param_space=(1e-10, 1.0, 10))
+clf = GaussianProcessClassification(kernel=kernel).fit(X, Y)
+
+# plot the decision function for each datapoint on the grid
+Z = clf.predict_proba(np.c_[xx.ravel(), yy.ravel()])
```

Comment:
You may want to use `np.concatenate`, `np.vstack(...).T` or `np.hstack` here for readability.